### ISSUE TITLE
Error with the Android SDK calling retrievePaymentIntent

### ISSUE THREAD

**Issue by zeulb** [stripe/stripe-terminal-react-native#722]
**Describe the bug**

Received this error after calling `retrievePaymentIntent` from `useStripeTerminal` on Android. No issue calling this method on iOS
```
{"error": {"code": "UNEXPECTED_ERROR.UNEXPECTED_SDK_ERROR", "message": "Unknown error retrieving payment intent from API"}, "paymentIntent": undefined}
```

**To Reproduce**

I'm using Stripe Connect connected account using test mode, we created the PI on the backend with

```
Stripe::PaymentIntent.create(
            {
              amount: @connect_invoice.amount_requested,
              currency: 'usd',
              customer: @connect_invoice.stripe_customer_id,
              description: @connect_invoice.description.gsub(/\s+/i, ' ').truncate(1000),
              metadata: {
                connect_invoice: @connect_invoice.id,
              },
              on_behalf_of: @connect_invoice.stripe_connect_account.account_id,
              payment_method_options: payment_method_options,
              payment_method_types: payment_method_types,
              statement_descriptor: StripePaymentIntent::DESCRIPTOR,
              transfer_data: {
                destination: @connect_invoice.stripe_connect_account.account_id,
              },
            },
          )
```

On the frontend we called `initialize()` and then `retrievePaymentIntent(secret)`

**Expected behavior**

**Screenshots**

**Stripe Terminal React Native SDK version**

0.0.1-beta.18

**Smartphone (please complete the following information):**

- Device: Samsung S21
- OS: Android 14.0

**Additional context**

**Comment by nazli-stripe** [comment]
Can you share a PaymentIntent ID for a failed payment?

**Comment by zeulb** [comment]
@nazli-stripe thanks for the swift response, here's the PaymentIntent ID `pi_3POaefIYC8jLo7F20FTWsBk0`

**Comment by nazli-stripe** [comment]
hey @zeulb we just released a new SDK version, `beta.19`. Can you upgrade and let us know if that fixes your issue? This is being caused by a native SDK bug when parsing some values for `payment_method_types `, and should have been fixed in Android 3.6.0, which the latest RN release is using

**Comment by zeulb** [comment]
I can confirm this is fixed for us now after upgrading, thank you! @nazli-stripe
